src/libawkward/builder/ArrayBuilder.cpp

```diff
@@ -1,493 +1,411 @@
-  }
+void ArrayBuilder::field_fast(const char *key) {
+  builder_.get()->field(key, false);
+}
 
-  void
-  ArrayBuilder::field_check(const char* key) {
-    builder_.get()->field(key, true);
-  }
+void ArrayBuilder::field_check(const char *key) {
+  builder_.get()->field(key, true);
+}
 
-  void
-  ArrayBuilder::field_check(const std::string& key) {
-    field_check(key.c_str());
-  }
+void ArrayBuilder::field_check(const std::string &key) {
+  field_check(key.c_str());
+}
 
-  void
-  ArrayBuilder::endrecord() {
-    maybeupdate(builder_.get()->endrecord());
-  }
+void ArrayBuilder::endrecord() { maybeupdate(builder_.get()->endrecord()); }
 
-  void
-  ArrayBuilder::maybeupdate(const BuilderPtr tmp) {
-    if (tmp  &&  tmp.get() != builder_.get()) {
-      builder_ = std::move(tmp);
-    }
+void ArrayBuilder::maybeupdate(const BuilderPtr tmp) {
+  if (tmp && tmp.get() != builder_.get()) {
+    builder_ = std::move(tmp);
   }
-
-  const char* ArrayBuilder::no_encoding = nullptr;
-  const char* ArrayBuilder::utf8_encoding = "utf-8";
 }
 
+const char *ArrayBuilder::no_encoding = nullptr;
+const char *ArrayBuilder::utf8_encoding = "utf-8";
+} // namespace awkward
+
 ////////// extern C interface
 
-uint8_t awkward_ArrayBuilder_length(void* arraybuilder,
-                                    int64_t* result) {
-  awkward::ArrayBuilder* obj =
-    reinterpret_cast<awkward::ArrayBuilder*>(arraybuilder);
+uint8_t awkward_ArrayBuilder_length(void *arraybuilder, int64_t *result) {
+  awkward::ArrayBuilder *obj =
+      reinterpret_cast<awkward::ArrayBuilder *>(arraybuilder);
   try {
     *result = obj->length();
-  }
-  catch (...) {
+  } catch (...) {
     return 1;
   }
   return 0;
 }
 
-uint8_t awkward_ArrayBuilder_clear(void* arraybuilder) {
-  awkward::ArrayBuilder* obj =
-    reinterpret_cast<awkward::ArrayBuilder*>(arraybuilder);
+uint8_t awkward_ArrayBuilder_clear(void *arraybuilder) {
+  awkward::ArrayBuilder *obj =
+      reinterpret_cast<awkward::ArrayBuilder *>(arraybuilder);
   try {
     obj->clear();
-  }
-  catch (...) {
+  } catch (...) {
     return 1;
   }
   return 0;
 }
 
-uint8_t awkward_ArrayBuilder_null(void* arraybuilder) {
-  awkward::ArrayBuilder* obj =
-    reinterpret_cast<awkward::ArrayBuilder*>(arraybuilder);
+uint8_t awkward_ArrayBuilder_null(void *arraybuilder) {
+  awkward::ArrayBuilder *obj =
+      reinterpret_cast<awkward::ArrayBuilder *>(arraybuilder);
   try {
     obj->null();
-  }
-  catch (...) {
+  } catch (...) {
     return 1;
   }
   return 0;
 }
 
-uint8_t awkward_ArrayBuilder_boolean(void* arraybuilder,
-                                     bool x) {
-  awkward::ArrayBuilder* obj =
-    reinterpret_cast<awkward::ArrayBuilder*>(arraybuilder);
+uint8_t awkward_ArrayBuilder_boolean(void *arraybuilder, bool x) {
+  awkward::ArrayBuilder *obj =
+      reinterpret_cast<awkward::ArrayBuilder *>(arraybuilder);
   try {
     obj->boolean(x);
-  }
-  catch (...) {
+  } catch (...) {
     return 1;
   }
```

Comment:
This is another thing that I would want to turn off. I'm assuming that this feature would also make

```c++
if (predicate_1) {
  consequent_1;
} else if (predicate_2) {
  consequent_2;
} else {
  alternate;
}
```

But that needs more room to breathe:

```c++
if (predicate_1) {
  consequent_1;
}
else if (predicate_2) {
  consequent_2;
}
else {
  alternate;
}
```

The latter takes up more lines, but all block-starting statements are now at the same level of depth and all closing brackets are equally visible.

Reply:
It's something people care about:

 - https://stackoverflow.com/q/65452234/1623645
 - https://stackoverflow.com/q/31976583/1623645

though neither of these are clang-format (vscode and Visual Studio).